GEOSldas_App/util/shared/python/read_GEOSldas.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+import struct
+import os
+import numpy as np
+
+def read_obs_param(fname):
+    print(f"Reading {fname}")
+
+    with open(fname, 'r') as fid:
+        N_obs_param = int(fid.readline().strip())
+
+        obs_param = []
+        for _ in range(N_obs_param):
+            param = {}
+            param['descr'] = fid.readline().strip().strip('"')
+            param['species'] = float(fid.readline().strip())
+            param['orbit'] = float(fid.readline().strip())
+            param['pol'] = float(fid.readline().strip())
+            param['N_ang'] = int(float(fid.readline().strip()))
+
+            param['ang'] = np.array([float(x) for x in fid.readline().split()])
+
+            param['freq'] = float(fid.readline().strip())
+            param['FOV'] = float(fid.readline().strip())
+            param['FOV_units'] = fid.readline().strip().strip('"')
+            param['assim'] = fid.readline().strip()
+            param['scale'] = fid.readline().strip()
+            param['getinnov'] = fid.readline().strip()
+            param['RTM_ID'] = float(fid.readline().strip())
+            param['bias_Npar'] = float(fid.readline().strip())
+            param['bias_trel'] = float(fid.readline().strip())
+            param['bias_tcut'] = float(fid.readline().strip())
+            param['nodata'] = float(fid.readline().strip())
+            param['varname'] = fid.readline().strip().strip('"')
+            param['units'] = fid.readline().strip().strip('"')
+            param['path'] = fid.readline().strip().strip('"')
+            param['name'] = fid.readline().strip().strip('"')
+
+            if 'Vv80' in fname or 'OL80' in fname:
```

Comment:
Not sure about this - I guess it is for backwards compatibility? I think any experiments run with the current release or later will be outputting maskpath and maskname here, so I've been using:
            param['maskpath'] = fid.readline().strip().strip('"')
            param['maskname'] = fid.readline().strip().strip('"')

Reply:
Right. The old file doesn't have 'maskpath' and 'maskname' and thus contains two fewer lines. The current code definitely needs some modification. I haven't given much thought to it.